update/Pos/Update/Poll/PollState.hs

```diff
@@ -0,0 +1,89 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+-- Types related Poll update system
+
+module Pos.Update.Poll.PollState
+       (
+         -- * Types
+         PollState (..)
+
+       -- * Lenses
+       -- * PollState
+       , psBlockVersions
+       , psAdoptedBV
+       , psSoftwareVersion
+       , psEpochProposers
+       , psConfirmedBVs
+       , psConfirmedProposals
+       , psActiveProposals
+       , psDelActivePropsIdx
+       , psSlottingData
+       , psFullRichmenData
+       , psIssuersStakes
+
+       -- * Functions
+       , modifyPollState
+       ) where
+
+import           Universum
+
+import           Control.Lens                 (makeLenses)
+import           Data.HashSet                 (HashSet)
+import qualified Data.HashMap.Strict          as HM
+
+import           Pos.Core.Types               (ApplicationName, BlockVersion,
+                                               BlockVersionData, EpochIndex,
+                                               NumSoftwareVersion, SoftwareVersion,
+                                               StakeholderId)
+import           Pos.Lrc.DB.Issuers           (IssuersStakes)
+import           Pos.Lrc.Types                (FullRichmenData)
+import           Pos.Slotting.Types           (SlottingData)
+import           Pos.Update.Core              (UpId)
+import           Pos.Update.Poll.Types        (BlockVersionState, ConfirmedProposalState,
+                                               PollModifier (..), ProposalState)
+import           Pos.Util.Modifier            (deletions, insertions)
+
+data PollState = PollState
+    { -- | All competing block versions with their states
+      _psBlockVersions      :: !(HashMap BlockVersion BlockVersionState)
+      -- | Presently adopted block version with its data
+    , _psAdoptedBV          :: !(BlockVersion, BlockVersionData)
+      -- | Software version of this node
+    , _psSoftwareVersion    :: !SoftwareVersion
```

Comment:
No, it's not needed. I see it's used in `setLastConfirmedSV` and `addConfirmedProposal`, but they should behave differently. Take a look at `MonadPoll PollT` implementation to see proper logic.